### ISSUE TITLE
Initializing component data when creating entities in 1.0 release

### ISSUE THREAD

**Issue by panuchka** [3mcd/javelin#168]
Hi! I have a question regarding component data initialization in 1.0 release. As I mentioned in the other issue (#167), I am refactoring my game to support the 1.0 version.

How can we, first of all, initialize an entity with a component that has some initial data. I presume this works a bit similarly to previous version:
```
world.create(
  component(Transform, 5, 0, 5),
  component(Actor, "character-basic"),
  component(Agent),
  component(Animation, "idle"),
  component(Follower)
);
```

Is this a proper way to initialize an entity still? For example, the `Transform` component with x,y,z values? How do they map to the component if the component is basically just a regular object?

The old way to create a component also had a separate `initialize` -function. Now that it is gone and components are just plain objects, what is the best way to add some initial data to the component object? Lets say I do not pass any data to the `Transform` -component, so I would like it to default to 0,0,0...

Thank you in advance and keep up the good work! 💪😎

**Comment by 3mcd** [comment]
Hey @panuchka, another good question. `component` accepts a second argument, an object, which will be `Object.assign`-ed to the newly created component after initialization. For example,

```ts
const velocity = component(Velocity, { x: 0, y: -9.81 })
velocity.x // 0
velocity.y // -9.81
```

If you frequently spawn lots of entities and garbage collection is a concern, you could also write a simple helper that would assign these fields without the need for an intermediate object that is thrown away immediately. But I felt the `initialize` function didn't add enough value to keep around, especially since I was aiming to simplify component schema definition as much as possible.

**Comment by panuchka** [comment]
Yes, this seems a lot simpler and clearer than the previous implementation! 🙂 Thank you for explaining it!
